gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetup.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks.setup;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Class responsible for installing the current JDK into {@code destinationJdkInstallationDir} and importing the
+ * certificates specified by their serialNumbers & alias name from {@code certsDir} into the JDK's truststore.
+ * A certificate will be imported iff the serial number already exists in the truststore.
+ * The class will be called by the Gradle setup script in
+ * <a href="file:../resources/gradle-jdks-setup.sh">resources/gradle-jdks-setup.sh</a>.
+ */
+public final class GradleJdkInstallationSetup {
+
+    public static void main(String[] args) throws IOException {
+        StdLogger logger = new StdLogger();
+        CaResources caResources = new CaResources(logger);
+        if (args.length != 2) {
+            throw new IllegalArgumentException("Expected two arguments: destinationJdkInstallationDir and certsDir");
+        }
+        Path destinationJdkInstallationDir = Path.of(args[0]);
+        Path certsDir = Path.of(args[1]);
+        atomicCopyJdkInstallationDirectory(logger, destinationJdkInstallationDir);
+        Map<String, String> certSerialNumbersToNames = extractCertsSerialNumbers(logger, certsDir);
+        caResources.maybeImportCertsInJdk(destinationJdkInstallationDir, certSerialNumbersToNames);
+    }
+
+    private static void atomicCopyJdkInstallationDirectory(ILogger logger, Path destinationJdkInstallationDirectory)
```

Comment:
I would just call this `copy` rather than `atomicCopy` - it's not actually atomic (it is locked though).